### PR TITLE
Added support for baseController attribute in Alloy xml tag

### PR DESCRIPTION
**Pull request by brentonhouse**
pull request for https://jira.appcelerator.org/browse/AC-642

### CODE REVIEW

**Comment by brentonhouse** [comment]
I signed CLA a long time ago... Not sure why it is giving this validation error.

**Comment by FokkeZB** [comment]
@brentonhouse probably because the [CLA server](https://cla.appcelerator.com) is down. We are replacing it this week by a new app and then continue to process your PR. Thanks!

**Comment by falkolab** [comment]
@brentonhouse,It is necessary PR, thx!

**Comment by feons** [comment]
@brentonhouse, great PR! We are ardently waiting for you to include a test case to make it awesome! :)

**Comment by brentonhouse** [comment]
Added Test App: https://github.com/brentonhouse/brentonhouse.alloy/tree/feature-basecontroller-attribute/test/apps/testing/ALOY-1360

File: Alloy/commands/compile/index.js [hunk 1]
```diff
@@ -706,7 +713,7 @@ function parseAlloyComponent(view, dir, manifest, noView, fileRestriction) {
 	}
 	var cCode = CU.loadController(files.CONTROLLER);
 	template.parentController = (cCode.parentControllerName !== '') ?
-		cCode.parentControllerName : "'BaseController'";
+    	cCode.parentControllerName : CU[CONST.DOCROOT_BASECONTROLLER_PROPERTY] || "'BaseController'";
```
Comment (feons) [inline Alloy/commands/compile/index.js]: Missing indentation.

Comment (brentonhouse) [inline Alloy/commands/compile/index.js]: Just to be clear, you want me to replace the spaces with a tab, correct?

Comment (feons) [inline Alloy/commands/compile/index.js]: Yes, the project favors tabs over spaces.
Also, line 716 is a continuation of the previous line, so 1 tab should be added to the current indentation.

Comment (brentonhouse) [inline Alloy/commands/compile/index.js]: I checked in a fix.  Let me know if that is not the correct format and I can resubmit.  Thanks!

**Comment by feons** [comment]
Looks great! **Approved.**

**Comment by FokkeZB** [comment]
@feons I cannot marge so go ahead ;)

**Pull request merged**